pkg/operator/operator.go

```diff
@@ -271,5 +274,47 @@ func (o *Operator) Config() *manifests.Config {
 		return manifests.NewDefaultConfig()
 	}
 
+	// Only fetch the the token and cluster ID if they have not been specified in the config.
+	if c.TelemeterClientConfig.ClusterID != "" && c.TelemeterClientConfig.Token != "" {
+		return c
+	}
+
+	cmap, err = o.client.KubernetesInterface().CoreV1().ConfigMaps("kube-system").Get("cluster-config-v1", metav1.GetOptions{})
+	if err != nil {
+		glog.V(4).Infof("Could not fetch cluster configuration from API. Proceeding without it.")
+		return c
+	}
+	ic := make(map[string]interface{})
+	if err := json.Unmarshal(cmap.BinaryData["install-config"], &ic); err != nil {
+		glog.V(4).Infof("Could not parse cluster configuration. Proceeding without it.")
+		return c
+	}
+	var ok bool
+	if c.TelemeterClientConfig.ClusterID == "" {
+		c.TelemeterClientConfig.ClusterID, ok = ic["clusterID"].(string)
+		if !ok {
+			glog.V(4).Infof("Could not parse cluster ID. Proceeding without it.")
+			return c
+		}
+	}
+	if c.TelemeterClientConfig.Token == "" {
+		ps := struct {
+			Auths struct {
+				COC struct {
+					Auth string `json:"auth"`
+				} `json:"cloud.openshift.com"`
+			} `json:"auths"`
+		}{}
```

Comment:
/cc @smarterclayton this is the current parsing logic for the pull secret

Reply:
ok, can we make sure tollbooth starts generating the secret in this form asap?